BLSQ/iaso
# IA-1873 UI for bulk user creation

Front-end needed to be added to the endpoint allowing to create users in bulk with a csv file

Related JIRA tickets : IA-1873

## Self proofreading checklist

- [X] Are my typescript files well typed-ish

## Changes

- Converted DownloadButtonsComponent to TS and exported the button styles
- Added an endpoint to download csv template
- Added Modal component + related buttons:
    - couldn't use `ConfirmCancelModal` as I needed to position the "Download template" button at flex-start (cs flex end for the usual Confirm and Cancel buttons) 
    - used backend validation hook to display error message in the modal). No translation yet available
    - Fixed a bug with the backend validation hook ( a `.join(',')` that made it fail silently)
    - Allow users to close the modal while the request is being handled since it can be long (ie: loading spinnner doens't cover the buttons)
    - The bulk create button is disabled when there is a mutation loading. This a broad-ish workaround to prevent users to upload multiple csv files while the backend is processing the current one. There is a loadiung spinner on the disabled button to indicate it's blocked because something is loading, and a tooltip with an explanation message.
    - Show the error snackbar, so users who closed the modal are still notified

## How to test

Go to Users
click "Create from file"
Download the template
Fill it out with several users. Note: org units should be referred to by id
Save. it should work
repeat with the same file. It should fail

## Print screen / video

Recording made on staging:
https://user-images.githubusercontent.com/38907762/223114556-e656b3d5-32c3-4630-b384-3d0ec0c47041.mov
